scout/server/blueprints/public/controllers.py

```diff
@@ -0,0 +1,161 @@
+import logging
+from functools import reduce
+
+from scout.constants import VERBS_MAP
+
+LOG = logging.getLogger(__name__)
+NOF_RECENT_EVENTS = 3
+
+
+"""Controller module for first webview.
+
+Requires: Python 3.7 or greater.
+Reason: Implementation relies on dicts being able to maintain internal order.
+This was previously only supported in OrderedDict().
+"""
+
+
+def get_events_of_interest(store, user):
+    """Read event database and compile a list of selected events of interest
+    Args:
+        store: scout.adapter.MongoAdapter
+        user: store.user
+    Returns:
+        events_of_interest: list of dicts{'case', 'link', 'human_readable'}
+        -where human_readable is a list of strings refering to users recent
+         events of interest
+    """
+    events_of_interest = []
+    events_per_case = []
+    cases = recent_cases(user, store)
+
+    for case in cases:
+        event_list = events_in_case(store, user, case)
+        events_per_case.append(event_list)
+
+    for event_list in events_per_case:
+        event = {}
+        event["human_readable"] = compile_important_events(event_list)
+        head, *_tail = event_list
+        event["link"] = head["link"]
+        event["case"] = head["case"]
+        events_of_interest.append(event)
+    return events_of_interest
+
+
+def recent_cases(user, store):
+    """Return a list of recent cases order in increasing age. A case
+    may appear only once."""
+    return list(store.unique_cases_by_date({"_id": user.email}))
+
+
+def events_in_case(store, user, case):
+    """Return a list of events associated with a user's specific case."""
+    return list(store.user_events_by_case({"_id": user.email}, case))
+
+
+def compile_important_events(event_list):
+    """Compile a string of recent events, ordered by importance, together
+    with the sum of the events occurences. Importance is determined by
+    placement in the macro VERBS_MAP.
```

Comment:
Would it be very complicated to display **when** the events occurred?

Reply:
The algorithm for displaying events is rewritten. It will now show the newest event first. If consecutive events are the same kind they will be aggregated. Date is "updated_at". It is displayed via a pretty print function.

![image](https://user-images.githubusercontent.com/1150065/167571976-73d1eec4-79d9-4b69-9fa2-02cea14d9a55.png)